dock/plugins/pre_add_yum_repo_by_url.py

```diff
@@ -0,0 +1,64 @@
+"""
+Add arbitrary yum repo, specified by URL of repo file, to a list of
+repos which should be injected into built image.
+
+This plugin has to run _BEFORE_ yum inject plugin.
+
+Example configuration to add content of repo file at URL:
+
+{
+    "name": "add_yum_repo_by_url",
+    "args": {
+        "repourls": ["http://example.com/myrepo/myrepo.repo"]
+    }
+}
+
+"""
+from dock.plugin import PreBuildPlugin
+import requests
+try:
+    # py2
+    from StringIO import StringIO
+    # use StringIO, not cStringIO
+    # cStringIO's readlines() gives bad results for unicode objects
```

Comment:
hmm, looks like worth reporting upstream, since cStringIO should be faster version of StringIO